broker/executor.go

```diff
@@ -0,0 +1,80 @@
+//  Copyright (c) 2017-2018 Uber Technologies, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+
+package broker
+
+import (
+	"github.com/uber/aresdb/query"
+	"github.com/uber/aresdb/datanode/topology"
+	"github.com/uber/aresdb/utils"
+	"github.com/uber/aresdb/memstore"
+	"net/http"
+)
+
+// QueryExecutor defines query executor
+type QueryExecutor interface {
+	// Execute executes query and flush result to connection
+	Execute(namespace, sqlQuery string, w http.ResponseWriter) (err error)
+}
+
+// NewQueryExecutor creates a new QueryExecutor
+func NewQueryExecutor(sm SchemaManager, topo topology.DynamicTopology) QueryExecutor {
```

Comment:
since we do not need to MarkShardsAvailable in broker, we can just use topology.Topology instead of DynamicTopology